test/test_series_io.py

```diff
@@ -81,8 +81,7 @@ def test_from_text_skip(tmpdir):
 def test_from_binary(tmpdir, eng):
     a = arange(8, dtype='int16').reshape((4, 2))
     p = os.path.join(str(tmpdir), 'data.bin')
-    with open(p, 'w') as f:
-        f.write(a)
+    a.tofile(p)
```

Comment:
Nope, that doesn't work in Python 3. It comes to this issues of `bytes` and `string` objects being different in Python 3. In Python 3, opening a file in `w` mode means that you have to pass it a string. When you pass it an `ndarray` instead it calls `ndarray.tostring()` which, in Python 2 returns a string (that encodes the bytes). But in Python 3 it returns a `bytes` object instead, which the file writer does not accept.

In hindsight, opening the file in `wb` mode would make the old code work in both Python 2 and 3. But since the `ndarray.tofile()` method does the right thing on both versions and is more concise, I vote we just leave it.